### PR TITLE
Fixing ESBJAVA-3915

### PR DESCRIPTION
**Pull request by Shakila**

### CODE REVIEW

File: components/mediation-ui/org.wso2.carbon.inbound.ui/src/main/java/org/wso2/carbon/inbound/ui/internal/InboundManagementClient.java [hunk 1]
```diff
@@ -183,7 +183,12 @@ public boolean addInboundEndpoint(String name, String sequence, String onError,
                     String tmpString = strKey + "." + i;
                     String strVal = prop.getProperty(tmpString);
                     if (strVal != null) {
-                        rtnList.add(strVal);
+                        if (strProtocol.equals(InboundClientConstants.TYPE_KAFKA) && (mandatory && !strVal.contains("highlevel.") && !strVal.contains("simple.")) || !mandatory) {
```
Comment (malakasilva) [inline components/mediation-ui/org.wso2.carbon.inbound.ui/src/main/java/org/wso2/carbon/inbound/ui/internal/InboundManagementClient.java]: Seems like issue with formatting. Line length

File: components/mediation-ui/org.wso2.carbon.inbound.ui/src/main/resources/web/inbound/inboundcommon.js [hunk 2]
```diff
@@ -263,3 +263,26 @@ function showAdvancedOptions(id) {
     }
 }
 
+function showSpecialFields(specialParams, inboundDescriptionOfParams) {
```
Comment (malakasilva) [inline components/mediation-ui/org.wso2.carbon.inbound.ui/src/main/resources/web/inbound/inboundcommon.js]: Can we simply have a span element and change visibility

File: components/mediation-ui/org.wso2.carbon.inbound.ui/src/main/java/org/wso2/carbon/inbound/ui/internal/InboundManagementClient.java [hunk 3]
```diff
@@ -361,4 +366,43 @@ public boolean updteInboundEndpoint(String name, String sequence, String onError
...
+        String specialParamsList = "";
+        loadProperties();
+        if (prop != null) {
+            String strKey = "kafka.mandatory";
+            String strLength = prop.getProperty(strKey);
+            Integer iLength = null;
+            if (strLength != null) {
+                try {
+                    iLength = Integer.parseInt(strLength);
+                } catch (Exception e) {
```
Comment (RKathees) [inline components/mediation-ui/org.wso2.carbon.inbound.ui/src/main/java/org/wso2/carbon/inbound/ui/internal/InboundManagementClient.java]: Specific NumberFormateException can be used to catch the exception while converting to integer

Comment (Shakila) [inline components/mediation-ui/org.wso2.carbon.inbound.ui/src/main/java/org/wso2/carbon/inbound/ui/internal/InboundManagementClient.java]: I used NumberFormateException at this point in another PR. https://github.com/wso2/carbon-mediation/pull/460

**Pull request merged**